src/ui/actions/app.ts

```diff
@@ -185,73 +186,109 @@ export function executeCommand(key: CommandKey): UIThunkAction {
+        dispatch(setSelectedPrimaryPanel("search"));
+        break;
+      case "open_function_search":
+        dispatch(setViewMode("dev"));
+        dispatch(openQuickOpen("@", true));
+        break;
+      case "open_network_monitor":
+        dispatch(setViewMode("dev"));
+        dispatch(setSelectedPanel("network"));
+        break;
+      case "open_print_statements":
+        dispatch(setViewMode("dev"));
+        dispatch(setSelectedPrimaryPanel("debugger"));
+        break;
+      case "open_react_devtools":
+        dispatch(setViewMode("dev"));
+        dispatch(setSelectedPanel("react-components"));
+        break;
+      case "open_sources":
+        dispatch(setViewMode("dev"));
+        dispatch(setSelectedPrimaryPanel("explorer"));
+        // Someday we'll fix circular dependencies. Today is not that day.
+        const { expandSourcesPane } = await import("devtools/client/debugger/src/actions/ui");
+        dispatch(expandSourcesPane());
+        break;
+      case "open_outline":
+        dispatch(setViewMode("dev"));
+        dispatch(setSelectedPrimaryPanel("explorer"));
+        userData.set("layout_debuggerOutlineExpanded", true);
+        break;
+      case "toggle_protocol_panel":
+        userData.toggle("feature_protocolPanel");
+        break;
+      case "toggle_protocol_timeline":
+        userData.toggle("feature_protocolTimeline");
+        break;
+      case "open_viewer":
+        dispatch(setViewMode("non-dev"));
+        break;
+      case "show_comments":
+        dispatch(setSelectedPrimaryPanel("comments"));
+        break;
+      case "show_console_filters":
+        dispatch(setViewMode("dev"));
+        dispatch(setSelectedPanel("console"));
+        break;
+      case "show_events":
+      case "show_replay_info":
+        dispatch(setSelectedPrimaryPanel("events"));
+        break;
+      case "show_privacy":
+        dispatch(setModal("privacy"));
+        break;
+      case "show_sharing":
+        dispatch(setModal("sharing", { recordingId }));
+        break;
+      case "toggle_edit_focus":
+        dispatch(toggleFocusMode());
+        break;
+      case "toggle_dark_mode":
+        let theme = userData.get("global_theme");
+        if (theme === "system") {
+          theme = getSystemColorScheme();
+        }
+        userData.set("global_theme", theme === "dark" ? "light" : "dark");
+        break;
+      case "pin_to_bottom":
+        dispatch(setToolboxLayout("bottom"));
+        break;
+      case "pin_to_left":
+        dispatch(setToolboxLayout("left"));
+        break;
+      case "pin_to_bottom_right":
+        dispatch(setToolboxLayout("ide"));
+        break;
+      case "jump_to_previous_pause":
+        dispatch(jumpToPreviousPause());
+        break;
+      case "jump_to_next_pause":
+        dispatch(jumpToNextPause());
+        break;
+      case "restart_session":
+        const url = setControllerKey(String(Date.now()));
+        window.location.assign(url);
+        break;
```

Comment:
adding this feature because i've found that it's helped me get out of cases where the backend freezes up and i want to keep debugging